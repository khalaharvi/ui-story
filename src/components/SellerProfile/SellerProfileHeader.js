import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import SectionHeaderTemplate, { cx } from 'templates/SectionHeader/SectionHeaderTemplate';
import { NewWindow as NewWindowIcon } from 'icons';
import { LinkWithIcon } from 'components/Link';

const SellerProfileHeader = ({ name, location, website }) => (
  <SectionHeaderTemplate title={name} subtitle={location} className={cx('-brand')}>
    {
      website &&
      <ul className={cx('brand-tools')}>
        <li className={classnames(cx('brand-tool'), cx('brand-tool--www'))}>
          <LinkWithIcon
            to={website}
            icon={<NewWindowIcon />}
            label="Official Website"
            external
          />
        </li>
      </ul>
    }
  </SectionHeaderTemplate>
);

SellerProfileHeader.propTypes = {
  name: PropTypes.string.isRequired,
  location: PropTypes.string.isRequired,
  website: PropTypes.string
};

export default SellerProfileHeader;
