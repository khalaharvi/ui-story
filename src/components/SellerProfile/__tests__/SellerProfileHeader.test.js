import React from 'react';
import { shallow } from 'enzyme';

import SellerProfileHeader from '../SellerProfileHeader';

describe('<SellerProfileHeader />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <SellerProfileHeader name="Name" location="Location" />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with website', () => {
    const wrapper = shallow(
      <SellerProfileHeader name="Name" location="Location" website="https://deringhall.com" />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
