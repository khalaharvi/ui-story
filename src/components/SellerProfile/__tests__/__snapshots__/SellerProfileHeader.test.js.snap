// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SellerProfileHeader /> renders correctly 1`] = `
<SectionHeaderTemplate
  className="dh-section-header--brand"
  subtitle="Location"
  title="Name"
/>
`;

exports[`<SellerProfileHeader /> renders correctly with website 1`] = `
<SectionHeaderTemplate
  className="dh-section-header--brand"
  subtitle="Location"
  title="Name"
>
  <ul
    className="dh-section-header-brand-tools"
  >
    <li
      className="dh-section-header-brand-tool dh-section-header-brand-tool--www"
    >
      <LinkWithIcon
        external={true}
        icon={<NewWindow />}
        label="Official Website"
        to="https://deringhall.com"
      />
    </li>
  </ul>
</SectionHeaderTemplate>
`;
